DESIGN.md

```diff
@@ -0,0 +1,36 @@
+# HTTP Provider Design
+
+The HTTP Provider offers functionality for interacting with generic HTTP servers as part of terraform configuration.
+Specifically, the provider issues an HTTP GET request to the defined URL on every Terraform run.
+
+Below we have a collection of _Goals_ and _Patterns_: they represent the guiding principles applied during the
+development of this provider. Some are in place, others are ongoing processes, others are still just inspirational.
+
+## Goals
+
+* [_Stability over features_](.github/CONTRIBUTING.md)
+* Provide managed data source to issue HTTP GET request.
```

Comment:
Is there IETF and/pr W3 RFCs to link to for this design choice? Which HTTP versions are supported? Are non-standard HTTP implementations supported (e.g. SPDY, QUIC)?

Reply:
Very good points.

Given the influx of requests, and the fact that once you start merging things here new requests for features are going to come, it's important to pin point which specific HTTP we support here.